students/AndrewKim/lesson06/mailroom_part4.py

```diff
@@ -0,0 +1,56 @@
+
+donor_history = {'Andrew Kim': [5.0, 3.0], 'Jamie Park': [4.0], 'Tim Duncan': [3.0], 'Billy Yan': [2.0]}
+
+def thank_you():
+    #prompt for a Full Name
+    name = input("Type the donor's full name: ")
+    #if user types 'list'
+    while name == "list":
+        for a in donor_history,keys():
+            print(a)
+        name = input("Type the donor's full name: ")
+
+    #if the name is not in the list
+    if name not in donor_history.keys():
+        donor_history[name] = []
+    
+    #add donation amount on the history
+    donation_amount = input("Please enter the donation amount: ")
+    try:
+        donor_history[name].append(float(donation_amount))
+        #print thank you
+        print(f'Thank you {name}, for your generous donation of {donation_amount}!') 
+    except ValueError:
+        print("Not a valid number! Please enter a valid number\n")
+    
```

Comment:
To make this function testable, I would refactor it to return the thank you letter as a string. Then you can print the string to the console wherever the function is called